core/src/main/java/org/mskcc/cbio/portal/scripts/ImportTypesOfCancers.java

```diff
@@ -32,74 +32,105 @@
-	        load(file, clobber);
-        } catch (RuntimeException e) {
-            throw e;
+            load(new File(filePath), clobber);
         } catch (IOException|DaoException e) {
             throw new RuntimeException(e);
         }
     }
 
     public static void load(File file, boolean clobber) throws IOException, DaoException {
-		SpringUtil.initDataSource();
-        if (clobber) DaoTypeOfCancer.deleteAllRecords(); //TODO - this option should not exist...in a relational DB it basically means the whole DB is cleaned-up...there should be more efficient ways to do this...and here it is probably an unwanted side effect. REMOVE??
-        TypeOfCancer aTypeOfCancer = new TypeOfCancer();
+        ProgressMonitor.setCurrentMessage("Loading cancer types...");
+        List<TypeOfCancer> typeOfCancerList = parseCancerTypesFromFile(file);
+        SpringUtil.initDataSource();
+        if (clobber) {
+            ProgressMonitor.setCurrentMessage("Deleting all previous cancer types...");
+            DaoTypeOfCancer.deleteAllRecords(); //TODO - remove this option - foreign key constraints may mean large cascade effects (possibly the deletion of all studies) - instead, change the option to 'deleteTypeOfCancerIfNotPresent' and add a loop through existing typeOfCancer records, removing those which are not in the parsed typeOfCancerList
+        }
+        writeRecordsToDatabase(typeOfCancerList, clobber);
+        ProgressMonitor.setCurrentMessage("Done.");
+        ConsoleUtil.showMessages();
+    }
+
+    private static List<TypeOfCancer> parseCancerTypesFromFile(File file) throws IOException {
+        ProgressMonitor.setCurrentMessage(String.format("Reading cancer types from file '%s'...", file.getPath()));
+        List<TypeOfCancer> typeOfCancerList = new ArrayList<TypeOfCancer>();
         Scanner scanner = new Scanner(file);
-        int numNewCancerTypes = 0;
-        
-        while(scanner.hasNextLine()) {
-            String[] tokens = scanner.nextLine().split("\t", -1);
-            if (tokens.length != 4) {
-                throw new IOException(
-                    "Cancer type file '" + file.getPath() +
-                    "' is not a five-column tab-delimited file");
-            }
-            
-            String typeOfCancerId = tokens[0].trim();
-            //if not clobber, then existing cancer types should be skipped:
-            if (!clobber && DaoTypeOfCancer.getTypeOfCancerById(typeOfCancerId.toLowerCase()) != null ) {
-            	ProgressMonitor.logWarning("Cancer type with id '" + typeOfCancerId + "' already exists. Skipping.");
+        while (scanner.hasNextLine()) {
+            String nextLine = scanner.nextLine();
+            String[] fields = nextLine.split("\t", -1);
+            if (fields.length != 4) {
+                throwBadColumnCountException(file, nextLine, fields);
             }
```

Comment:
@sheridancbio maybe good to add expectedFieldCount as argument to this function and move the `fields.length!=4` to be inside the function. Could rename the function to e.g. `checkColumnCount`. Currently we still have `4` and `four` in two places which is what caused the inconsistent message before

Reply:
ok, will do